app/commands/git/sync_pull_requests.rb

```diff
@@ -0,0 +1,163 @@
+module Git
+  class SyncPullRequests
+    include Mandate
+
+    def call
+      repos.each do |repo|
+        @repo = repo
+        begin
+          sync!
+        rescue StandardError => e
+          Rails.logger.error "Error syncing pull requests for #{repo}: #{e}"
+        end
+      end
+    end
+
+    private
+    attr_reader :repo, :pull_requests, :cursor
+
+    def repos
+      page = 1
+      fetched = []
+
+      loop do
+        response = octokit_client.search_repositories("org:exercism is:public", page: page, per_page: 100)
+        fetched += response[:items].map { |item| item[:full_name] }
+        page += 1
+
+        break fetched if fetched.size >= response[:total_count]
+      end
+    end
+
+    def sync!
+      fetch!
+      create!
+    end
+
+    def create!
+      pull_requests.each do |pr|
+        ::GithubPullRequest.create_or_find_by!(github_id: pr[:pull_request][:id]) do |p|
+          p.github_username = pr[:pull_request][:user][:login]
+          p.github_repo = pr[:repository][:full_name]
+          p.github_event = pr
+        end
+      end
+    end
+
+    def fetch!
+      @cursor = nil
+      @pull_requests = []
+
+      loop do
+        page_data = fetch_page!
+        @pull_requests += pull_requests_from_response(page_data)
+
+        page_info = page_data[:data][:repository][:pullRequests][:pageInfo]
+        @cursor = page_info[:endCursor]
```

Comment:
Again, not a huge fan of the mutation of an instance variable here. I'd rather pass the cursor into `fetch_page!`.

Reply:
That was what my first version did. I somehow thought this was more Ruby-esque. 🤷